azure-pipelines.perf-test.yml

```diff
@@ -10,86 +10,140 @@ variables:
-
-  - task: Bash@3
-    inputs:
-      filePath: yarn-ci.sh
-    displayName: yarn
-
-  # @fluentui/perf-test-northstar needs build and bundle steps
-  - script: |
-      yarn lage build --to @fluentui/perf-test-northstar --to perf-test --verbose
-    displayName: build to perf-test
-
-  # Fluent perf-test must run before Fabric perf-test until they are consolidated.
-  # @fluentui/perf-test-northstar Baseline
-  - script: |
-      yarn perf:test:base
-    condition: eq(variables.isPR, false)
-    workingDirectory: packages/fluentui/perf-test-northstar
-    displayName: Perf Test Base (Fluent)
-
-  # @fluentui/perf-test-northstar PR
-  - script: |
-      yarn perf:test
-    condition: eq(variables.isPR, true)
-    workingDirectory: packages/fluentui/perf-test-northstar
-    displayName: Perf Test (Fluent)
-
-  # Fabric: perf-test will build, bundle, and run-perftest
-  - script: |
-      yarn just perf-test
-    workingDirectory: apps/perf-test
-    displayName: Perf Test
-
-  - task: AzureUpload@2
-    displayName: Upload Perf Test Result to PR deploy site (Fluent N*)
-    inputs:
-      SourcePath: 'packages/fluentui/perf-test-northstar/dist'
-      azureSubscription: $(azureSubscription)
-      storage: $(azureStorage)
-      ContainerName: '$web'
-      BlobPrefix: '$(deployBasePath)/perf-test-northstar'
-
-  - task: AzureUpload@2
-    displayName: Upload Perf Test Result to PR deploy site (Fluent)
-    inputs:
-      SourcePath: 'apps/perf-test/dist'
-      azureSubscription: $(azureSubscription)
-      storage: $(azureStorage)
-      ContainerName: '$web'
-      BlobPrefix: '$(deployBasePath)/perf-test'
-
-  - task: GithubPRComment@0
-    displayName: 'Post Perf Results to Github Pull Request'
-    inputs:
-      githubOwner: microsoft
-      githubRepo: 'fluentui'
-      blobFilePath: '$(Build.SourcesDirectory)/$(PerfCommentFilePath)'
-      status: '$(PerfCommentStatus)'
-      uniqueId: 'perfComment9423'
-
-  - script: |
-      yarn stats:build
-    condition: eq(variables.isPR, false)
-    displayName: Bundle Statistics (master only)
-
-  - script: |
-      yarn perf
-    condition: eq(variables.isPR, false)
-    displayName: Performance Tests (master only)
-
-  # HEADS UP: also see tag-version-prefix in fluentui-publish.js
-  - script: |
-      yarn stats:save --tag=`git tag --points-at HEAD | grep ^@fluentui/react-northstar_v | grep -o 'northstar_v.*'`
-    condition: eq(variables.isPR, false)
-    displayName: Save Statistics to DB (master only)
-    env:
-      STATS_URI: $(STATS_URI)
-
-  - template: .devops/templates/cleanup.yml
+jobs:
+  - job: CheckIfPackagesAffected
+    steps:
+      - template: .devops/templates/tools.yml
+
+      - task: Bash@3
+        inputs:
+          filePath: yarn-ci.sh
+        displayName: yarn
+
+      - script: |
+          yarn check-if-packages-affected
+        name: PackagesAffected
+        displayName: Check if v8 and/or  northstar packages were affected
+
+  - job: PerfTestNorthstar
```

Comment:
this whole conditional checks via CI capabilities is interesting...

what I had in mind was to leverage task orchestration tool we use at the moment (lage) to not do "things twice"

meaning:

```sh
lage perf:test  --since master
```

would do all the work for us (determining what needs to build and what not and execute test accordingly )

then I guess we would still need to imperatively handle uploading to Azure

Reply:
It does seem like this will ultimately still require some form of condition on whether to upload results to Azure or not which has been the main issue with this PR 😅. Also, I'm a little confused by what you mean by not do "things twice"? I don't believe there's any repetition of builds happening here at all.  My understanding of the performance issue for the perf-test pipeline is that it unnecessarily built and ran both northstar and v8 perf-test packages when only either and sometimes neither `@fluentui/react` or `@fluentui/react-northstar` were even affected by a PR. If both packages are affected, their respective perf-test packages will still need to be built and have tests execute which is what it's currently doing now by default.

Reply:
> Also, I'm a little confused by what you mean by not do "things twice"

1. lage tells you what to do (what is affected, what to build etc)
2. we have do this checks "by hand" in pipeline configuration and manually trigger

> If both packages are affected, their respective perf-test packages will still need to be built and have tests execute which is what it's currently doing now by default.

yes that's all correct.


We should tackle this approach (`lage perf:test  --since master`) as a follow up. 

From DX perspective, ideally only one consistent approach should be taken regarding what needs to be build and how. ATM we have logic like this 
![image](https://user-images.githubusercontent.com/1223799/155502185-02bc8a74-f041-49a1-b0b7-d1ad60dc984b.png). Now if that would change, one needs to be aware that such a config exists and you need to change it. if we use lage/nx this would not be needed as the tool understand the dependency tree thus trigger correct executions.